bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/themes/ColorsAndFontsPreferencePage.java

```diff
@@ -331,9 +334,22 @@ private class PresentationLabelProvider extends LabelProvider implements IFontPr
 
 		private int usableImageSize = -1;
 
+		private final Runnable updateControlsAndRefreshTreeRunnable = new Runnable() {
+
+			@Override
+			public void run() {
+				if (fontChangeButton == null || fontChangeButton.isDisposed()) {
+					return;
+				}
+				updateControls();
+				tree.getViewer().refresh();
+			}
+		};
+
 		private IPropertyChangeListener listener = event -> {
 			if (event.getNewValue() != null) {
 				fireLabelProviderChanged(new LabelProviderChangedEvent(PresentationLabelProvider.this));
+				Display.getDefault().timerExec(300, updateControlsAndRefreshTreeRunnable);
```

Comment:
what's this magic number 300 about?

Reply:
I simply chose a small delay of 300ms to run a refresh in the UI. A lot change events are thrown and I wanted to ensure that the UI is not refreshed on each event, but bundled in the end with one call. Is there an alternative easy way to realize this? Or is there a delay constant representing my intent?

Reply:
Maybe use org.eclipse.jface.util.Throttler?

Reply:
Throttler also requires a magic number for the second parameter minWaitTime. The magic number would not disappear here if Throttler instead of Display.getDefault().timerExec would be used.
Sorry, I don't see the benefit yet. Do you see an advantage? Can you explain?

Reply:
I agree. I was just wondering why 300 and not 200 or 400. ;-)